src/openrct2/platform/shared.c

```diff
@@ -554,10 +580,73 @@ void platform_process_messages()
 	}
 
 	gCursorState.any = gCursorState.left | gCursorState.middle | gCursorState.right;
+}
+
+static void platform_process_text_input_special_keys(keypress key)
+{
+	if (key.keycode == SDLK_BACKSPACE) {
+		if (key.mod & KEYBOARD_PRIMARY_MODIFIER) {
+			// Clear the input on <CTRL>Backspace (Windows/Linux) or <MOD>Backspace (macOS)
+			textinputbuffer_clear(&gTextInput);
+			console_refresh_caret();
+			window_update_textbox();
+
+			// If we have input text with a cursor position none zero
+		} else if (gTextInput.selection_offset > 0) {
+			size_t endOffset = gTextInput.selection_offset;
+			textinputbuffer_cursor_left(&gTextInput);
+			gTextInput.selection_size = endOffset - gTextInput.selection_offset;
+			textinputbuffer_remove_selected(&gTextInput);
+
+			console_refresh_caret();
+			window_update_textbox();
+		}
+	}
+
+	if (key.keycode == SDLK_HOME) {
+		textinputbuffer_cursor_home(&gTextInput);
+		console_refresh_caret();
+	}
 
-	// Updates the state of the keys
-	sint32 numKeys = 256;
-	gKeysState = SDL_GetKeyboardState(&numKeys);
+	if (key.keycode == SDLK_END) {
+		textinputbuffer_cursor_end(&gTextInput);
+		console_refresh_caret();
+	}
+
+	if (key.keycode == SDLK_DELETE) {
+		size_t startOffset = gTextInput.selection_offset;
+		textinputbuffer_cursor_right(&gTextInput);
+		gTextInput.selection_size = gTextInput.selection_offset - startOffset;
+		gTextInput.selection_offset = startOffset;
+		textinputbuffer_remove_selected(&gTextInput);
+		console_refresh_caret();
+		window_update_textbox();
+	}
+
+	if (key.keycode == SDLK_RETURN) {
+		window_cancel_textbox();
+	}
+
+	if (key.keycode == SDLK_LEFT) {
+		textinputbuffer_cursor_left(&gTextInput);
+		console_refresh_caret();
+
+	} else if (key.keycode == SDLK_RIGHT) {
+		textinputbuffer_cursor_right(&gTextInput);
+		console_refresh_caret();
+
+	} else if (platform_keypress_equals(key, (keypress){ SDLK_v, KEYBOARD_PRIMARY_MODIFIER })) {
```

Comment:
Might be a dump question, but do all platforms only allow <kbd>cmd/ctrl</kbd>+<kbd>v</kbd> as paste shortcut?